ut/alarm_tables_testing.cpp

```diff
@@ -0,0 +1,321 @@
+}
+
+TEST_F(SNMPTest, ActiveTableIndex)
+{
+  // Creates three custom alarm definitions and raises the alarms.
+  AlarmDef::SeverityDetails cleared1(AlarmDef::CLEARED, "First alarm cleared description", "First alarm cleared details");
+  AlarmDef::SeverityDetails raised1(AlarmDef::CRITICAL, "First alarm raised description", "First alarm raised details");
+  AlarmDef::AlarmDefinition example1(6666, AlarmDef::SOFTWARE_ERROR, {cleared1, raised1});
+  AlarmTableDef def1_cleared(example1, cleared1);
+  AlarmTableDef def1_raised(example1, raised1);
+  alarmActiveTable_trap_handler(def1_raised);
+
+  AlarmDef::SeverityDetails cleared2(AlarmDef::CLEARED, "Second alarm cleared description", "Second alarm cleared details");
+  AlarmDef::SeverityDetails raised2(AlarmDef::CRITICAL, "Second alarm raised description", "Second alarm raised details");
+  AlarmDef::AlarmDefinition example2(6667, AlarmDef::SOFTWARE_ERROR, {cleared2, raised2});
+  AlarmTableDef def2_cleared(example2, cleared2);
+  AlarmTableDef def2_raised(example2, raised2);
+  alarmActiveTable_trap_handler(def2_raised);
+
+  AlarmDef::SeverityDetails cleared3(AlarmDef::CLEARED, "Third alarm cleared description", "Third alarm cleared details");
+  AlarmDef::SeverityDetails raised3(AlarmDef::CRITICAL, "Third alarm raised description", "Third alarm raised details");
+  AlarmDef::AlarmDefinition example3(6668, AlarmDef::SOFTWARE_ERROR, {cleared3, raised3});
+  AlarmTableDef def3_cleared(example3, cleared3);
+  AlarmTableDef def3_raised(example3, raised3);
+  alarmActiveTable_trap_handler(def3_raised);
+
+  // Shells out to snmpwalk to find all entries in Alarm Active Table.
+  std::vector<std::string> entries = snmp_walk(alarmActiveTable_table);
+
+  
+  // Retrieves the current date and time.
+  struct timespec ts;
+  struct tm timing;
+  time_t rawtime;
+  clock_gettime(CLOCK_REALTIME, &ts);
+  rawtime = ts.tv_sec;
+  timing = *localtime(&rawtime);
+  int day = timing.tm_mday;
+  int month = 1 + timing.tm_mon;
+
+  // Splits each part of the index field (separated by ".") for the first alarm into a vector.
+  std::vector<std::string> index_fields1;
+  Utils::split_string(entries[0], '.', index_fields1);
+
+  // As defined in RFC3877 the 17th and 16th elements of the index field are the day and
+  // month that the alarm was raised respectively. This checks those values are
+  // the current day and month as retreived above. 
+  ASSERT_EQ(std::to_string(day), index_fields1[17]);
+  ASSERT_EQ(std::to_string(month), index_fields1[16]);
+ 
+  // Checks the index of the first alarm contains 26 elements as defined in RFC
+  // 3877.
+  ASSERT_EQ(26, index_fields1.size());
+  
+  // Retreives each alarm's index (which should be a strictly monotonically
+  // increasing integer which indexes the entries within the table and is the
+  // 26th and last element of the index field) from the index field of the table.
+  // As the alarm's index is immediately proceeded by a space and equality
```

Comment:
It might be clearer to have a small helper function `std::string get_index_from_row(std::string snmpwalk_row)` which did this.